components/scripts/gradle/gradle-init-scripts/configure-gradle-enterprise.gradle

```diff
@@ -12,7 +12,7 @@ initscript {
 
     def getInputParam = { String name ->
         def envVarName = name.toUpperCase().replace('.', '_').replace('-', '_')
-        return System.getProperty(name) ?: System.getenv(envVarName)
+        return gradle.startParameter.systemPropertiesArgs[name] ?: System.getenv(envVarName)
```

Comment:
Add a comment to this method explaining that the way we read system properties here means they are not tracked as inputs to the configuration cache model. The same is the case for reading environment variables with Gradle [6.5, 7.3].